plugins/generator-1.20.1/forge-1.20.1/templates/structure/structure.java.ftl

```diff
@@ -99,7 +87,7 @@ public class ${name}Feature extends Feature<NoneFeatureConfiguration> {
 				</#if>
 
 				<#if data.restrictionBlocks?has_content>
-				if (!base_blocks.contains(context.level().getBlockState(new BlockPos(i, j, k)).getBlock()))
+				if (!(${containsAnyOfBlocks(data.restrictionBlocks "context.level().getBlockState(new BlockPos(i, j, k))" data.excludeRestrictionBlocks)}))

```

Comment:
You could probably negate excludeRestrictionBlocks and remove ! from if to avoid double negations